packages/react-client/src/ReactFlightClient.js

```diff
@@ -2714,9 +2756,35 @@ function resolveChunkDebugInfo(
   chunk: SomeChunk<any>,
 ): void {
   if (__DEV__ && enableAsyncDebugInfo) {
-    // Push the currently resolving chunk's debug info representing the stream on the Promise
-    // that was waiting on the stream.
-    chunk._debugInfo.push({awaited: streamState._debugInfo});
+    // Add the currently resolving chunk's debug info representing the stream
+    // to the Promise that was waiting on the stream, or its underlying value.
+    const debugInfoEntry: ReactAsyncInfo = {awaited: streamState._debugInfo};
+
+    const addDebugInfo = () => {
```

Comment:
We shouldn't create the closure outside the condition. Maybe it's optimized but we should make sure.

Reply:
But it's used in both branches of the condition.

Reply:
In that case you should move it out to a reusable function in module scope and then call that from the closure. So that the closure only needs to be allocated in the delayed branch.

Reply:
Then `addDebugInfo` can be hoisted out.